nifi-nar-bundles/nifi-graph-bundle/nifi-other-graph-services/src/main/java/org/apache/nifi/graph/TinkerpopClientService.java

```diff
@@ -0,0 +1,538 @@
+            int port = context.getProperty(PORT).evaluateAttributeExpressions().asInteger();
+            String path = context.getProperty(PATH).evaluateAttributeExpressions().getValue();
+            String[] contactPoints = contactProp.split(",[\\s]*");
+            for (String contactPoint : contactPoints) {
+                builder.addContactPoint(contactPoint.trim());
+            }
+            builder.port(port).path(path);
+        } else {
+            File yamlFile = new File(context.getProperty(REMOTE_OBJECTS_FILE).evaluateAttributeExpressions().getValue());
+            try {
+                builder = Cluster.build(yamlFile);
+            } catch (Exception ex) {
+                throw new ProcessException(ex);
+            }
+        }
+        builder = setupSSL(context, builder);
+
+        if (context.getProperty(USER_NAME).isSet() && context.getProperty(PASSWORD).isSet()) {
+            String username = context.getProperty(USER_NAME).evaluateAttributeExpressions().getValue();
+            String password = context.getProperty(PASSWORD).evaluateAttributeExpressions().getValue();
+            builder.credentials(username, password);
+        }
+
+        return builder.create();
+    }
+
+    protected Map<String, String> scriptSubmission(String query, Map<String, Object> parameters, GraphQueryResultCallback handler) {
+        try {
+            Client client = cluster.connect();
+            Iterator<Result> iterator = client.submit(query, parameters).iterator();
+            long count = 0;
+            while (iterator.hasNext()) {
+                Result result = iterator.next();
+                Object obj = result.getObject();
+                if (obj instanceof Map) {
+                    handler.process((Map)obj, iterator.hasNext());
+                } else {
+                    handler.process(new HashMap<>() {{
+                        put("result", obj);
+                    }}, iterator.hasNext());
+                }
+                count++;
+            }
+
+            Map<String, String> resultAttributes = new HashMap<>();
+            resultAttributes.put(NODES_CREATED, NOT_SUPPORTED);
+            resultAttributes.put(RELATIONS_CREATED, NOT_SUPPORTED);
+            resultAttributes.put(LABELS_ADDED, NOT_SUPPORTED);
+            resultAttributes.put(NODES_DELETED, NOT_SUPPORTED);
+            resultAttributes.put(RELATIONS_DELETED, NOT_SUPPORTED);
+            resultAttributes.put(PROPERTIES_SET, NOT_SUPPORTED);
+            resultAttributes.put(ROWS_RETURNED, String.valueOf(count));
+
+            return resultAttributes;
+
+        } catch (Exception ex) {
+            throw new ProcessException(ex);
+        }
+    }
+
+    protected Map<String, String> bytecodeSubmission(String s, Map<String, Object> map, GraphQueryResultCallback graphQueryResultCallback) {
+        String hash = DigestUtils.md5Hex(s);
```

Comment:
MD5 is not a secure hashing function. Although this is not a cryptographic function, recommend using SHA-256.

Reply:
I think MD5 is fine for this because the goal is to just make it quick and easy to uniquely identify compiled scripts.

Reply:
I think moving from md5 will help with security findings when deployed within a secure environment. So moving to SHA-256.